### ISSUE TITLE
improvements to the donor data input flow

### ISSUE THREAD

**Issue by zagrodzki** [zagrodzki/labdoo#47]
Currently the physical cleaning crew copies data from the collective (batch) donor sheet to the individual paper process sheets (with a pen), and PXE stations copy data from the process slip to the input forms. We should develop a process to not have to copy-paste that data anymore.

**Comment by zagrodzki** [comment]
As an example from today:

      1         "donor_name": "Pädag Maturität Kreuylingen",                    
      1         "donor_name": "Pädag Maturität Kreuzlingen",                    
      1         "donor_name": "Pädagogische Maturitatsschule Kreuzlingen",      
      2         "donor_name": "Pädagogische Maturitätsschule Kreutzlingen",     
      4         "donor_name": "Pädagogische Maturitätsschule Kreuzlingen",

**Comment by Ronald65** [comment]
I suggest, that the process slip are prepared from Labdoo entry process. So cleaning crew dont have to enter any data by pen. I am working on an excel sheet with pulldown menues, where donor can be entered. 
Maybe you have a ohter idea.... ??

**Comment by zagrodzki** [comment]
Yes, we should not put that data in an Excel file - that makes it difficult to have more than one person work simultaneously on new entries and would again bring us the problems related to power outages etc.

A better option would beonline spreadsheet (like Office365 online version of Excel or Google Sheets). That could work, but will require some preparations (e.g. the spreadsheet should validate the checksum in the laptop ID) and will not allow the system to generate JSON files for labdoo.org updates automatically - the JSON file would need to be updated from the collected spreadsheet first.

The best option seems to make adding and selecting donors a part of the installation server.
Batch donations could be either added in the system directly before the event, or could be uploaded through a CSV file. During the event, the laptop entry would first be created during after physical cleaning by connecting sticker ID and donor/donation data (this would configure the laptop with status S1). At the PXE station the setup process and quality check would update the data in the system with technical details, updating the status to S2 or other depending on quality check results.

**Comment by zagrodzki** [comment]
If that last option sounds in theory acceptable, I can prepare some mockups of how the interface would look like and we can discuss from there. Let me know.

**Comment by zagrodzki** [comment]
After my discussions with Ron last Tuesday:

The current donor data entry process:

![current process](https://docs.google.com/drawings/d/e/2PACX-1vQehHDKJM5AIeMBNTH3quxx00bGnmvx5Gawg3m9DqAb03ju9FrWgMlgnsf2L809ypjqDr9d_T77rby-/pub?w=960&h=720)

Donor data (marked in green) is a lot of text where it's easy to make mistakes, so we want to reduce the places where it needs to be copied manually (solid arrows).

Proposed process:

![proposed](https://docs.google.com/drawings/d/e/2PACX-1vTJY2Dpo_CCcbbesn2Hm2nP-XPsamE5cpUIQj5Jc6X7FE8e9mbB-Q17jFyyvraJh5Ja_zp7jnhBkgV4/pub?w=960&h=720)

Donation ID (bright green) is short and checksummed, so it's easy to copy without making mistakes.

To make donation ID visibly different from laptop ID, the donation ID will start with a letter prefix (e.g. "D" or "DB") and the checksum will also be a letter. We'll use the same checksum schema as laptop IDs, but we'll convert the checksum digit into a letter - A=0, B=1 etc.

So where previously laptop entry form would look like this:

```
Laptop ID (NNNN-N): [123-4              ]
Donor name:         [sebek              ]
Donor email:        [sebek@zagrodzki.net]
Collection date:    [01.06.2019         ]
Collected by:       [Brack.ch           ]
```

It will now look like this:

```
Laptop ID (nnnn-n):      [123-4       ]
Donation ID (DB-nnnn-x): [DB-123-E    ]
````
